### DISCUSSION TITLE
Timezone settings prevent data from being written to Prometheus - how to fix it?

### DISCUSSION THREAD

**Discussion by vivierer** [ccfos/nightingale#1297] [Q&A]
因为把n9ev5、promethous和telegraf安装在了一台主机上，主机的时区时cst，promethous的时区时utc，在使用telegraf的exec插件调用自己写的脚本读取信息时遇到了时区不匹配导致promethous丢弃数据的情况。
telegraf调取的其他插件都可以把数据提交给n9e，然后再存入promethous。只有exec调用脚本反馈的监测数据可以提交给n9e，但是n9e提交给promethous时因为差了8个小时无法存入promethous。
当通过配置n9e的server.conf把时间强制指为服务器时，能好半个小时，这半个小时数据可以进入promethous，也可以在n9e中查到，半个小时候，又无法存入promethous。此时查看promethous的日志信息，发现又因为时区原因丢弃了数据。
请问如何解决呀。

**Comment by UlricQin** [comment]
你用exec写的脚本，输出的内容是influx格式？influx格式可以不指定时间戳，这样exec采集的数据就会复用telegraf所在机器的当前时间，这样的话，就和cpu、mem那些指标使用了一样的时间戳了

**Comment by vivierer** [comment]
我修改了n9e的配置文件，使用了utc解决了。但是发现了新问题。telegraf n9e prometheus日志中都没有报错了。但是exec取得的数据依旧只能存储30分左右。我使用telegraf --debug，返回有exec取得的数据，但是就是查询不到。

**Comment by vivierer** [comment]
我输出的是influx格式，也没有指定时间戳。最后通过调整server.conf里面的实验参数解决了问题。谢谢




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年12月2日(星期五) 凌晨0:56
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ccfos/nightingale] 测试出一个因为时区设置导致数据无法存入prometheus的小问题，请问怎么解决 (Discussion #1297)





 
你用exec写的脚本，输出的内容是influx格式？influx格式可以不指定时间戳，这样exec采集的数据就会复用telegraf所在机器的当前时间，这样的话，就和cpu、mem那些指标使用了一样的时间戳了
